nuxeo-features/nuxeo-search/nuxeo-search-core/src/main/java/org/nuxeo/ecm/platform/search/core/SavedSearchWriter.java

```diff
@@ -88,9 +92,22 @@ protected void writeEntityBody(SavedSearch search, JsonGenerator jg) throws IOEx
             jg.writeStringField(param, search.getNamedParams().get(param));
         }
 
+        // write own schema properties as search parameters (unprefixed)
+        SchemaManager schemaManager = Framework.getService(SchemaManager.class);
+        DocumentModel doc = search.getDocument();
+        Set<String> ownSchemas = new HashSet<>(Arrays.asList(doc.getSchemas()));
+        doc.getFacets().stream()
+                .map(facet -> schemaManager.getFacet(facet).getSchemas())
+                .flatMap(Collection::stream).forEach(schema -> ownSchemas.remove(schema.getName()));
+        for (String schema : ownSchemas) {
+            writeSchemaProperties(jg, doc, schema, false);
+        }
+        jg.writeEndObject();
+
+        // write document model properties
         Set<String> schemas = ctx.getProperties();
         if (schemas.size() > 0) {
-            DocumentModel doc = search.getDocument();
+            jg.writeObjectFieldStart("properties");
```

Comment:
API contract is modified, right? We should probably do this change behind the `v11` API and keep the previous contract for `v1`. We can continue this discussion on [#team-platform](https://nuxeo.slack.com/messages/CJVB4LKT3).